contracts/PFAUnit.sol

```diff
@@ -11,27 +11,30 @@
 pragma solidity >=0.8.0 <0.9.0;
 
 import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
 import "./PFA.sol";
 import "./libraries/CodeVerification.sol";
 import "./libraries/Immutable.sol";
-import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "./interfaces/IERC20Payable.sol";
 
 /// @title Standard pay-for-access (PFA) contract. Also implements
 /// ERC-721 standard (G_NFT).
 /// @author brandon@formless.xyz
-contract PFAUnit is PFA, ERC721 /* G_NFT */ {
+contract PFAUnit is IERC20Payable, ERC721, PFA {
     /// @notice Emitted when a payment is sent to the owner of this
     /// PFA.
     event PaymentToOwner(address indexed owner, uint256 value);
 
     string public constant NAME = "SHARE";
     string public constant SYMBOL = "PFA";
     uint256 private constant UNIT_TOKEN_INDEX = 0;
-
     string internal _tokenURI;
 
+    // ERC-20 contract address e.g. USDC
+    address private erc20ContractAddress;
```

Comment:
nit: By convention in our codebase, for private Solidity variables let's use the underscore prefix, eg. `_erc20ContractAddress`.